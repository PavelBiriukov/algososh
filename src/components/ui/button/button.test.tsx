import { fireEvent, render } from "@testing-library/react"
import { Button } from "./button"

describe('Button component', () => {
    test('rendering: buttons with text', () => {
      const { asFragment } = render(<Button text="unit test"/>);
      expect(asFragment()).toMatchSnapshot();
    });
    test('rendering: buttons without text', () => {
        const { asFragment } = render(<Button/>);
        expect(asFragment()).toMatchSnapshot();
    });
    test('rendering: the locked button', () => {
        const { asFragment } = render(<Button disabled/>);
        expect(asFragment()).toMatchSnapshot();
    });
    test('rendering: buttons with loading indication', () => {
        const { asFragment } = render(<Button isLoader/>);
        expect(asFragment()).toMatchSnapshot();
    });
    test('We check the correctness of the callback call when clicking on the button', () => {
        const handleClick = jest.fn();
        const { getByText } = render(<Button text="button" onClick={handleClick} />);
        const button = getByText('button');
        fireEvent.click(button);
        expect(handleClick).toHaveBeenCalledTimes(1);
    });
});