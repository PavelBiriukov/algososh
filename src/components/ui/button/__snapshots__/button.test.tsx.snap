// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component rendering: buttons with loading indication 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined loader "
    disabled=""
    type="button"
  >
    <img
      alt="Загрузка."
      class="loader_icon"
      src="loader.svg"
    />
  </button>
</DocumentFragment>
`;

exports[`Button component rendering: buttons with text 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    >
      unit test
    </p>
  </button>
</DocumentFragment>
`;

exports[`Button component rendering: buttons without text 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</DocumentFragment>
`;

exports[`Button component rendering: the locked button 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined false "
    disabled=""
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</DocumentFragment>
`;
